pkg/reconciler/events/cloudevent/cloudevent_test.go

```diff
@@ -209,3 +240,62 @@ func TestEventForPipelineRun(t *testing.T) {
 		})
 	}
 }
+
+func TestEventForRun(t *testing.T) {
+	runTests := []struct {
+		desc          string
+		run           *v1alpha1.Run
+		wantEventType TektonEventType
+	}{{
+		desc:          "send a cloud event with unset condition, just started",
+		run:           getRunByCondition("", ""),
+		wantEventType: RunStartedEventV1,
+	}, {
+		desc:          "send a cloud event with unknown status pipelinerun, empty reason",
+		run:           getRunByCondition(corev1.ConditionUnknown, ""),
+		wantEventType: RunRunningEventV1,
+	}, {
+		desc:          "send a cloud event with unknown status pipelinerun, some reason set",
+		run:           getRunByCondition(corev1.ConditionUnknown, "custom controller reason"),
+		wantEventType: RunRunningEventV1,
+	}, {
+		desc:          "send a cloud event with successful status pipelinerun",
+		run:           getRunByCondition(corev1.ConditionTrue, "yay"),
+		wantEventType: RunSuccessfulEventV1,
+	}, {
+		desc:          "send a cloud event with unknown status pipelinerun",
+		run:           getRunByCondition(corev1.ConditionFalse, "meh"),
+		wantEventType: RunFailedEventV1,
+	}}
+
+	for _, c := range runTests {
+		t.Run(c.desc, func(t *testing.T) {
+			names.TestingSeed()
```

Comment:
I see `names.TestingSeed` a lot in our tests-- what's the reason it's needed here, just out of curiosity?

Reply:
Copy paste? I wondered the same but I lazy :P I will try to find out

Reply:
its there for a reason, I have been wondering about it as well since I started writing unit tests.

It was introduced to use `fixed` seed for the generated names as part of PR https://github.com/tektoncd/pipeline/pull/491. This PR fixed the issue where it was reported the resource names were too long https://github.com/tektoncd/pipeline/issues/481.

@afrittoli some of these might no longer be needed since now we have fixed names for some of the resources as your implemented in https://github.com/tektoncd/pipeline/pull/4361.

I generally comment out one line at a time to see if its breaking my test if I am not sure what that code is doing 😄 Or add the same call twice 🙃 to understand the impact.

Reply:
Thanks @pritidesai - good idea :)

`names.TestingSeed` is used to initialise the random generator used in names, so that you can get know what name to expect to validate results in tests. As @pritidesai mentioned, the need for this is probably reduced now that children resources have a more predictable name. 

CloudEvent tests only match event types, so the name seed is not strictly required. It does cost anything or do any harm either, but I'll remove it as it's not needed.